src/plugins/auto-block-recovery/is-invalid.js

```diff
@@ -396,6 +401,33 @@ export const getTagTree = html => {
 	}, [] )
 }
 
+/**
+ * Checks whether the block contains empty id attributes
+ * e.g. <div id=""></div>
+ *
+ * @param {Array} issue The invalidation array
+ *
+ * @return {boolean} if true, the html content has empty id attributes. Otherwise, false.
+ */
+export const hasEmptyId = issue => {
+	if ( ! issue.args ) {
+		return false
+	}
+
+	if ( issue.args.length !== 5 ) {
+		return false
+	}
+
+	// Block contents
+	if ( typeof issue.args[ 3 ] !== 'string' || typeof issue.args[ 4 ] !== 'string' ) {
+		return false
+	}
+
+	const oldHasEmptyId = ! issue.args[ 3 ].match( /id=""/ )
+	const newHasEmptyId = ! issue.args[ 4 ].match( /id=""/ )
+	return oldHasEmptyId && ! newHasEmptyId
```

Comment:
A be confusing. If `oldHasEmptyId` is `true`, it should mean that it has an empty id.

Also, if it's just a straight string match, it might be better to do it without regex.